### PR TITLE
chore(manager): add more type annotations

### PR DESCRIPTION
**Pull request by ikesyo**
Ref: #4312, #4301

### CODE REVIEW

**Comment by ViceIce** [comment]
Ref: #1261

File: lib/manager/bazel/extract.ts [hunk 1]
```diff
@@ -4,7 +4,9 @@ import { parse as _parse } from 'url';
 import { logger } from '../../logger';
 import { PackageDependency, PackageFile } from '../common';
 
-function parseUrl(urlString: string) {
+function parseUrl(
+  urlString: string
+): { repo: string; currentValue: string } | null {
```
Comment (ViceIce) [inline lib/manager/bazel/extract.ts]: Shouldn't we add a type or interface here?

File: lib/manager/bazel/update.ts [hunk 2]
```diff
@@ -40,9 +40,12 @@ function extractUrls(content: string) {
   return urls;
 }
 
-async function getHashFromUrl(url: string) {
+async function getHashFromUrl(url: string): Promise<string | null> {
   const cacheNamespace = 'url-sha256';
-  const cachedResult = await renovateCache.get(cacheNamespace, url);
+  const cachedResult: string | null = await renovateCache.get(
```
Comment (ViceIce) [inline lib/manager/bazel/update.ts]:
```suggestion
  const cachedResult = await renovateCache.get<string | null>(
```

File: lib/manager/common.ts [hunk 3]
```diff
@@ -173,15 +173,17 @@ export interface ManagerApi {
...
 
   extractPackageFile?(
     content: string,
     packageFile?: string,
     config?: ExtractConfig
-  ): Result<PackageFile>;
+  ): Result<PackageFile | null>;
 
-  getPackageUpdates(config: PackageUpdateConfig): Result<PackageUpdateResult[]>;
+  getPackageUpdates?(
```
Comment (ViceIce) [inline lib/manager/common.ts]: @rarkins Is `getPackageUpdates` optional?

File: lib/manager/gradle-wrapper/update.ts [hunk 4]
```diff
@@ -35,11 +35,11 @@ export async function updateDependency(
   }
 }
 
-function replaceType(url: string) {
+function replaceType(url: string): string {
   return url.replace('bin', 'all');
 }
 
-async function getChecksum(url: string) {
+async function getChecksum(url: string): Promise<any> {
```
Comment (ViceIce) [inline lib/manager/gradle-wrapper/update.ts]: We should add a type here.

File: lib/manager/homebrew/extract.ts [hunk 5]
```diff
@@ -98,7 +98,13 @@ function extractUrl(content: string) {
...
 }
 
-export function parseUrlPath(urlStr: string) {
+export function parseUrlPath(
+  urlStr: string
+): {
+  currentValue: string;
+  ownerName: string;
+  repoName: string;
+} | null {
```
Comment (ViceIce) [inline lib/manager/homebrew/extract.ts]: Add type or interface

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/manager/common.ts (hunk 3 above)
Comment (rarkins) [reply lib/manager/common.ts]: Yes, I think so

File: lib/manager/gradle-wrapper/update.ts (hunk 4 above)
Comment (ikesyo) [reply lib/manager/gradle-wrapper/update.ts]: Okay, let's assume `response.body` as `string` here.

File: lib/manager/common.ts (hunk 3 above)
Comment (ikesyo) [reply lib/manager/common.ts]: Only `lib/manager/travis` implements this.

File: lib/manager/homebrew/extract.ts (hunk 5 above)
Comment (ikesyo) [reply lib/manager/homebrew/extract.ts]: Something like this?

```ts
interface UrlPathParsedResult {
  currentValue: string;
  ownerName: string;
  repoName: string;   
}
```

File: lib/manager/bazel/extract.ts (hunk 1 above)
Comment (ikesyo) [reply lib/manager/bazel/extract.ts]: Something like this?

```ts
export interface UrlParsedResult {
  repo: string;
  currentValue: string
}
```

**Review by viceice — APPROVED** [review]
LGTM

**Pull request merged**